FrejaEidClient/src/main/java/com/verisec/frejaeid/client/http/HttpService.java

```diff
@@ -168,6 +174,63 @@ public final <Response extends FrejaHttpResponse> Response send(String methodUrl
         }
     }
 
+    @Override
+    public final HttpResponse httpGet(String methodUrl, Map<String, String> parameters)
+            throws FrejaEidClientInternalException, FrejaEidException, UnsupportedEncodingException {
+
+        HttpResponse httpResponse = null;
+        HttpGet request = null;
+        HttpStatusCode httpStatusCode = null;
+
+        URIBuilder uriBuilder = new URIBuilder();
+        uriBuilder.setPath(methodUrl);
+        for (Map.Entry<String, String> parameter : parameters.entrySet()) {
+            uriBuilder.addParameter(parameter.getKey(), URLEncoder.encode(parameter.getValue(),
+                                                                          StandardCharsets.UTF_8.toString()));
+        }
+
+        try {
+            request = new HttpGet(uriBuilder.build());
+            request.addHeader("Content-Type", "application/json");
+            request.addHeader(HttpHeaders.USER_AGENT, userAgentHeader);
+            httpResponse = httpClient.execute(request);
+            LOG.debug("Successfully sent HttpGet request to address {}.", methodUrl);
+            int httpStatusCodeValue = httpResponse.getStatusLine().getStatusCode();
+            String responseString =  EntityUtils.toString(httpResponse.getEntity(), StandardCharsets.UTF_8);
+            httpStatusCode = HttpStatusCode.getHttpStatusCode(httpStatusCodeValue);
+            if (httpStatusCode == null) {
+                throw new FrejaEidException(
+                        String.format("Received unsupported HTTP status code %s. Received HTTP message: %s.",
+                                      httpResponse.getStatusLine().getStatusCode(), responseString));
+            }
+            switch (httpStatusCode) {
+                case OK:
+                case NO_CONTENT:
+                    return httpResponse;
```

Comment:
I strongly suggest having only one exit point from the method, otherwise it's very hard to follow all potential flows of execution.

Reply:
Done, no_content is now treated as an error and throws an exception as it is not the expected result.